In [5]:
import pandas as pd

### Reading the dataset

In [6]:
asthma_data = pd.read_csv('../data/asthma_disease_data.csv')

In [7]:
asthma_data

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0             0.894448     5.488696      8.701003  ...          1.369051   
1             5.897329     6.341014      5.153966  ...          2.197767   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
4             4.604493     3.127048      9.625799  ...          3.470589   
...                ...          ...           ...  ...               ...   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2388          5.805180     4.386992      7.731192  ...          1.132977   
2389          4.735169     8.214064      7.483521  ...          1.685962   
2390          9.672637     7.362861      6.717272  ...          3.481549   
2391          1.613138     7.412878      8.512253  ...          2.280613   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            4.941206         0                  0               1         0   
1            1.702393         1                  0               0         1   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
4            3.067944         1                  1               1         0   
...               ...       ...                ...             ...       ...   
2387         5.166032         0                  1               0         0   
2388         5.509502         0                  0               0         1   
2389         3.346877         1                  0               1         1   
2390         1.713274         0                  1               1         0   
2391         2.453284         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                     0                1          0       Dr_Confid  
1                     1                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
4                     0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2387                  0                1          1       Dr_Confid  
2388                  1                0          1       Dr_Confid  
2389                  0                1          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  
2391                  1                0          0       Dr_Confid  

[2392 rows x 29 columns]

### Metadata

#### About dataset
This dataset contains extensive health information for 2,392 patients diagnosed with Asthma Disease, uniquely identified with IDs ranging from 5034 to 7425. It includes demographic details, lifestyle factors, environmental and allergy factors, medical history, clinical measurements, symptoms, and a diagnosis indicator. This dataset is valuable for researchers and data scientists aiming to explore factors associated with Asthma, develop predictive models, and conduct statistical analyses.

#### Demographic details
**Age**: The age of the patients ranges from 5 to 80 years.

**Gender**: Gender of the patients, where **0 represents Male** and **1 represents Female**.

**Ethnicity**: The ethnicity of the patients, coded as follows:
0: Caucasian
1: African American
2: Asian
3: Other

**EducationLevel**: The education level of the patients, coded as follows:
0: None
1: High School
2: Bachelor's
3: Higher

#### Lifestyle factors
**BMI**: Body Mass Index of the patients, ranging from 15 to 40.

**Smoking**: Smoking status, where **0 indicates No** and **1 indicates Yes**.

**PhysicalActivity**: Weekly physical activity in hours, ranging from 0 to 10.

**DietQuality**: Diet quality score, ranging from 0 to 10.

**SleepQuality**: Sleep quality score, ranging from 4 to 10.

#### Environmental and allergy factors
**PollutionExposure**: Exposure to pollution, score from 0 to 10.

**PollenExposure**: Exposure to pollen, score from 0 to 10.

**DustExposure**: Exposure to dust, score from 0 to 10.

**PetAllergy**: Pet allergy status, where 0 indicates No and 1 indicates Yes.

#### Medical history
**FamilyHistoryAsthma**: Family history of asthma, where 0 indicates No and 1 indicates Yes.
    
**HistoryOfAllergies**: History of allergies, where 0 indicates No and 1 indicates Yes.
    
**Eczema**: Presence of eczema, where 0 indicates No and 1 indicates Yes.
    
**HayFever**: Presence of hay fever, where 0 indicates No and 1 indicates Yes.
    
**GastroesophagealReflux**: Presence of gastroesophageal reflux, where 0 indicates No and 1 indicates Yes.

#### Clinical measurements
**LungFunctionFEV1**: Forced Expiratory Volume in 1 second (FEV1), ranging from 1.0 to 4.0 liters.

**LungFunctionFVC**: Forced Vital Capacity (FVC), ranging from 1.5 to 6.0 liters.

#### Symptoms
**Wheezing**: Presence of wheezing, where 0 indicates No and 1 indicates Yes.

**ShortnessOfBreath**: Presence of shortness of breath, where 0 indicates No and 1 indicates Yes.

**ChestTightness**: Presence of chest tightness, where 0 indicates No and 1 indicates Yes.
    
**Coughing**: Presence of coughing, where 0 indicates No and 1 indicates Yes.
    
**NighttimeSymptoms**: Presence of nighttime symptoms, where 0 indicates No and 1 indicates Yes.
    
**ExerciseInduced**: Presence of symptoms induced by exercise, where 0 indicates No and 1 indicates Yes.


#### Diagnosis info
**Diagnosis**: Diagnosis status for Asthma, where 0 indicates No and 1 indicates Yes.

### Observations

In [8]:
asthma_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2392 entries, 0 to 2391
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   PatientID               2392 non-null   int64  
 1   Age                     2392 non-null   int64  
 2   Gender                  2392 non-null   int64  
 3   Ethnicity               2392 non-null   int64  
 4   EducationLevel          2392 non-null   int64  
 5   BMI                     2392 non-null   float64
 6   Smoking                 2392 non-null   int64  
 7   PhysicalActivity        2392 non-null   float64
 8   DietQuality             2392 non-null   float64
 9   SleepQuality            2392 non-null   float64
 10  PollutionExposure       2392 non-null   float64
 11  PollenExposure          2392 non-null   float64
 12  DustExposure            2392 non-null   float64
 13  PetAllergy              2392 non-null   int64  
 14  FamilyHistoryAsthma     2392 non-null   

Loaded data corresponds to the info in dataset metadata.

There are 2392 observations with no missing values.

Except in 'DoctorInCharge', values of all other columns have numerical data type. Some are categorical variables represented as integers where 0 indicates 'no' and 1 indicates 'yes'. Genders are represented by 0 for male and 1 for female. Other variables are continuous and have float64 dtype. All details are described in the table and metadata info above.